doc/ss_logger.rst

```diff
@@ -0,0 +1,201 @@
+
+SmartSim offers **four** log functions to use within the Python driver script. The
+below functions accept string messages:
+
+- ``logger.error()``
+- ``logger.warning()``
+- ``logger.info()``
+- ``logger.debug()``
+
+The `SMARTSIM_LOG_LEVEL` environment variable accepts **four** log levels: `quiet`,
+`info`, `debug` and `developer`. Setting the log level in the environment (or via the override function)
+controls the log messages that are output at runtime. The log levels are listed below from
+highest severity to lowest severity:
+
+- level: `quiet`
+   - The `quiet` log level instructs SmartSim to print ``error()`` and ``warning()`` messages.
+- level: `info`
+   - The `info` log level instructs SmartSim to print ``info()``, ``error()`` and ``warning()`` messages.
+- level: `debug`
+   - The `debug` log level instructs SmartSim to print ``debug()``, ``info()``, ``error()`` and ``warning()`` messages.
+- level: `developer`
+   - The `developer` log level instructs SmartSim to print ``debug()``, ``info()``, ``error()`` and ``warning()`` messages.
+
+.. note::
+    The `developer` log level is intended for use during code development and signifies highly detailed and verbose logging.
+
+.. note::
+    `SMARTSIM_LOG_LEVEL` defaults to log level `info`. For SmartSim log sample usage examples, continue to the Configuration section.
+
+=============
+Configuration
+=============
+-------------
+Log to stdout
+-------------
+.. _log_to_stdout:
+The ``get_logger()`` function in SmartSim enables users to initialize a logger instance.
+Once initialized, a user may use the instance to log a message using one of the four
+logging functions.
+
+To use the SmartSim logger within a Python script, import the required `get_logger`
+function from the `log` module:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger
+
+Next, initialize an instance of the logger and provide a logger `name`:
+
+.. code-block:: python
+
+      logger = get_logger("SmartSim")
+
+To demonstrate full functionality of the SmartSim logger, we include all log
+functions in the Python driver script with log messages:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Execute the script *without* setting the `SMARTSIM_LOG_LEVEL`. Remember that `SMARTSIM_LOG_LEVEL`
+defaults to `info`. When we execute the script, the following messages will print to stdout::
+    11:15:00 system.host.com SmartSim[130033] INFO This is a message
+    11:15:00 system.host.com SmartSim[130033] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[130033] WARNING This is a warning message
+
+Notice that the `debug` function message was filtered. This is because by using
+a higher severity level (`info`), we instruct SmartSim to omit the lower severity levels (`debug` and `developer`).
```

Comment:
Notice that the `debug` function message was filtered. This is because by using
a higher severity level (`info`), we instruct SmartSim to omit the lower severity levels (`debug` and `developer`).

->

Notice that the `debug` function message was filtered. This is because by using
a lower verbosity level (`info`), we instruct SmartSim to omit the higher verbosity level messages (`debug` and `developer`).